src/timeline/RoomListDBusInterface.cpp

```diff
@@ -0,0 +1,89 @@
+// SPDX-FileCopyrightText: 2022 Nheko Contributors
+//
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#include "RoomListDBusInterface.h"
+
+#include "Cache_p.h"
+#include "ChatPage.h"
+#include "Logging.h"
+#include "MainWindow.h"
+#include "MxcImageProvider.h"
+#include "RoomlistModel.h"
+
+#include <QDBusConnection>
+
+RoomListDBusInterface::RoomListDBusInterface(RoomlistModel *parent)
+  : QObject{parent}
+  , m_parent{parent}
+{}
+
+QVector<nheko::dbus::RoomInfoItem>
+RoomListDBusInterface::getRooms(const QDBusMessage &message)
+{
+    // Just in case a room is joined while the model is being built.
+    auto modelSize = m_parent->models.size();
+    auto model{new QVector<nheko::dbus::RoomInfoItem>};
```

Comment:
```suggestion
    auto rooms = m_parent->models;
    auto model = std::make_shared<QVector<nheko::dbus::RoomInfoItem>>();
```

And adapt the code below to that